### DISCUSSION TITLE
Next.js behaviour I'm confused about ---

### DISCUSSION THREAD

**Discussion by gr-qft** [vercel/next.js#19203] [Help]
@lfades  @timneutkens  @leerob 

Please help, I'm really confused now. (Note: this does not happen when running `yarn dev`, it only happens when doing `yarn build` and then `yarn start`).

I'm following along this example:

https://github.com/vercel/next.js/tree/canary/examples/with-firebase-authentication/

where I'm using the component `FirebaseAuth` inside 

https://github.com/vercel/next.js/tree/canary/examples/with-firebase-authentication/component

exactly as it's shown there. I have a login page that exports a Login component, like below:

```js
export default function Login() {
  return (
    <div className={loginStyle}>
      <div>
        <FirebaseAuth />
      </div>
    </div>
  );
}
```

This page is linked from the main page. If I click the link from the main page I can get to the login page and the auth component renders normally. But if I go to the browser and type out the full address (e.g. localhost:3000/login) then I got the following error

```
Uncaught Error: Could not find the FirebaseUI widget element on the page.
```
I've spent hours on this and I'm completely confused now. Please help. Thanks!

**Comment by leerob** [comment]
If you clone and start the example without making any changes, does it work? That should help you figure out what's wrong with your code, since the example should be working 😄

**Reply by gr-qft** [reply]
It works, but I still don't understand why it's not working for my app. I've tried different things and it seems that removing line 51 in the code below makes my app work. 

https://github.com/vercel/next.js/blob/1e3534e1694f2c59b32025357039cd006d6e3532/examples/with-firebase-authentication/utils/auth/useUser.js#L49-L54 

I'm still trying to understand why, but things are getting more confusing for me. I think the key point to focus on is why everything works in dev but not after the app is built (the error message only shows up in the built version). Is there something basic about Next.js that I don't understand? 

Thanks!

**Reply by gr-qft** [reply]
I think I know the reason why the error message. I use the hook in the navigation component so the hook is called on every page and it always triggers a redirect whichever page visited. What I still don't understand is why it works on dev but not on built. Any idea as to why that's the case? 

Thanks!
